llvm/lib/Analysis/LazyValueInfo.cpp

```diff
@@ -1283,6 +1285,20 @@ std::optional<ValueLatticeElement> LazyValueInfoImpl::getValueFromICmpCondition(
   return ValueLatticeElement::getOverdefined();
 }
 
+std::optional<ValueLatticeElement>
+LazyValueInfoImpl::getValueFromTrunc(Value *Val, TruncInst *Trunc,
+                                     bool IsTrueDest) {
+  assert(Trunc->getType()->isIntegerTy(1));
```

Comment:
Could you please add a test with a vector select that has a trunc condition? I think it will fail this assertion. This should be isIntOrIntVectorTy.

Reply:
assert fixed and test added see `and_elide_trunc_cond_vec`